kloppy.helpers.transform does not allow the expected inputs

According to the method definition, the `kloppy.helpers.transform` method should allow three optional attributes: `to_orientation`, `to_pitch_dimensions` and `to_coordinate_system`. However, the implementation requires that (1) either `to_pitch_dimensions` or `to_coordinate_system` should be given when `to_orientation` is not None (I do not see why that would be required) and (2) `to_pitch_dimensions` does not allow a `PitchDimensions` object (it only works with a list or tuple).

## Minimal example 

```python
from kloppy.domain import (AttackingDirection, DatasetFlag, Dimension, Frame,
                           Ground, Metadata, MetricaCoordinateSystem,
                           Orientation, Period, PitchDimensions, Player,
                           PlayerData, Point, Point3D, Provider, Team,
                           TrackingDataset)

home_team = Team(team_id="home", name="home", ground=Ground.HOME)
away_team = Team(team_id="away", name="away", ground=Ground.AWAY)
teams = [home_team, away_team]

periods = [
    Period(
        id=1,
        start_timestamp=0.0,
        end_timestamp=10.0,
        attacking_direction=AttackingDirection.HOME_AWAY,
    ),
    Period(
        id=2,
        start_timestamp=15.0,
        end_timestamp=25.0,
        attacking_direction=AttackingDirection.AWAY_HOME,
    ),
]

metadata = Metadata(
    flags=~(DatasetFlag.BALL_OWNING_TEAM | DatasetFlag.BALL_STATE),
    pitch_dimensions=PitchDimensions(
        x_dim=Dimension(0, 100), y_dim=Dimension(-50, 50)
    ),
    orientation=Orientation.HOME_TEAM,
    frame_rate=25,
    periods=periods,
    teams=teams,
    score=None,
    provider=None,
    coordinate_system=None,
)

dataset = TrackingDataset(
    metadata=metadata,
    records=[
        Frame(
            frame_id=1,
            timestamp=0.1,
            ball_owning_team=teams[0],
            ball_state=None,
            period=metadata.periods[0],
            players_data={},
            other_data=None,
            ball_coordinates=Point3D(x=100, y=-50, z=0),
        ),
    ],
)

```

Only changing the orientation does not work. You also have to provide the pitch dimensions or coordinate system.
```python
transformed_dataset = dataset.transform(
        to_orientation=Orientation.FIXED_HOME_AWAY,
)
>>> E       AttributeError: 'NoneType' object has no attribute 'pitch_dimensions'
>>> kloppy/domain/services/transformers/dataset.py:50: AttributeError
```

You have to provide the pitch dimensions as a list or a tuple. If you provided them as a `PitchDimensions` object, they are simply ignored.
```python
transformed_dataset = dataset.transform(
        to_pitch_dimensions=PitchDimensions(
        x_dim=Dimension(0, 1), y_dim=Dimension(0, 1)
    ),
)
assert transformed_dataset.frames[0].ball_coordinates == Point3D(x=0, y=1, z=0)
>>>       assert transformed_dataset.frames[0].ball_coordinates == Point3D(x=0, y=1, z=0)
>>> E       AssertionError: assert Point3D(x=100, y=-50, z=0) == Point3D(x=0, y=1, z=0)
```

--- a/kloppy/domain/services/transformers/dataset.py
+++ b/kloppy/domain/services/transformers/dataset.py
@@ -1,6 +1,6 @@
-from dataclasses import asdict, fields, replace
+from dataclasses import fields, replace
 from kloppy.domain.models.tracking import PlayerData
-from typing import TypeVar, Union
+from typing import Union, Optional
 
 from kloppy.domain import (
     AttackingDirection,
@@ -8,7 +8,6 @@ from kloppy.domain import (
     DatasetFlag,
     EventDataset,
     Frame,
-    Metadata,
     Orientation,
     PitchDimensions,
     Point,
@@ -16,7 +15,6 @@ from kloppy.domain import (
     Team,
     TrackingDataset,
     CoordinateSystem,
-    Origin,
 )
 from kloppy.domain.models.event import Event
 from kloppy.exceptions import KloppyError
@@ -25,12 +23,12 @@ from kloppy.exceptions import KloppyError
 class DatasetTransformer:
     def __init__(
         self,
-        from_coordinate_system: CoordinateSystem = None,
-        from_pitch_dimensions: PitchDimensions = None,
-        from_orientation: Orientation = None,
-        to_coordinate_system: CoordinateSystem = None,
-        to_pitch_dimensions: PitchDimensions = None,
-        to_orientation: Orientation = None,
+        from_coordinate_system: Optional[CoordinateSystem] = None,
+        from_pitch_dimensions: Optional[PitchDimensions] = None,
+        from_orientation: Optional[Orientation] = None,
+        to_coordinate_system: Optional[CoordinateSystem] = None,
+        to_pitch_dimensions: Optional[PitchDimensions] = None,
+        to_orientation: Optional[Orientation] = None,
     ):
 
         if (
@@ -40,23 +38,46 @@ class DatasetTransformer:
             and to_coordinate_system
         ):
             raise ValueError(
-                "You can't specify both a PitchDimension and CoordinateSysetm transformation on the same Transformer instance"
+                "You can't specify both a PitchDimension and CoordinateSystem transformation on the same Transformer instance"
             )
 
         self._from_coordinate_system = from_coordinate_system
-        self._from_pitch_dimensions = (
-            from_pitch_dimensions
-            if from_pitch_dimensions
-            else from_coordinate_system.pitch_dimensions
-        )
-        self._from_orientation = from_orientation
+        if from_pitch_dimensions:
+            self._from_pitch_dimensions = from_pitch_dimensions
+        elif from_coordinate_system:
+            self._from_pitch_dimensions = (
+                from_coordinate_system.pitch_dimensions
+            )
+        else:
+            raise ValueError(
+                "You must either specify the source PitchDimension or CoordinateSystem"
+            )
+
         self._to_coordinate_system = to_coordinate_system
-        self._to_pitch_dimensions = (
-            to_pitch_dimensions
-            if to_pitch_dimensions
-            else to_coordinate_system.pitch_dimensions
-        )
+        if to_pitch_dimensions:
+            if from_pitch_dimensions is None:
+                raise ValueError(
+                    "You must specify the source PitchDimension when specifying the target PitchDimension"
+                )
+            self._to_pitch_dimensions = to_pitch_dimensions
+        elif to_coordinate_system:
+            if from_coordinate_system is None:
+                raise ValueError(
+                    "You must specify the source CoordinateSystem when specifying the target CoordinateSystem"
+                )
+            self._to_pitch_dimensions = to_coordinate_system.pitch_dimensions
+
+        self._from_orientation = from_orientation
         self._to_orientation = to_orientation
+        if (
+            from_orientation
+            and not to_orientation
+            or not from_orientation
+            and to_orientation
+        ):
+            raise ValueError(
+                "You must specify both the source and target Orientation"
+            )
 
     @property
     def _needs_coordinate_system_change(self):
@@ -64,16 +85,11 @@ class DatasetTransformer:
 
     @property
     def _needs_pitch_dimensions_change(self):
-        if self._from_coordinate_system and self._to_coordinate_system:
-            return (
-                self._from_coordinate_system.pitch_dimensions
-                != self._to_coordinate_system.pitch_dimensions
-            )
+        return self._from_pitch_dimensions != self._to_pitch_dimensions
 
-        if self._from_pitch_dimensions and self._to_pitch_dimensions:
-            return self._from_pitch_dimensions != self._to_pitch_dimensions
-
-    def change_point_dimensions(self, point: Union[Point, Point3D]) -> Point:
+    def change_point_dimensions(
+        self, point: Union[Point, Point3D, None]
+    ) -> Union[Point, Point3D, None]:
 
         if point is None:
             return None
@@ -89,7 +105,9 @@ class DatasetTransformer:
         else:
             return Point(x=x, y=y)
 
-    def flip_point(self, point: Union[Point, Point3D]):
+    def flip_point(
+        self, point: Union[Point, Point3D, None]
+    ) -> Union[Point, Point3D, None]:
 
         if not point:
             return None
@@ -112,12 +130,16 @@ class DatasetTransformer:
         self,
         ball_owning_team: Team,
         attacking_direction: AttackingDirection,
-        action_executing_team: Team = None,
+        action_executing_team: Optional[Team] = None,
     ) -> bool:
-        if self._from_orientation == self._to_orientation:
+        if (
+            self._from_orientation is None
+            or self._to_orientation is None
+            or self._from_orientation == self._to_orientation
+        ):
             flip = False
         else:
-            if not action_executing_team:
+            if action_executing_team is None:
                 action_executing_team = ball_owning_team
 
             orientation_factor_from = (
@@ -209,17 +231,15 @@ class DatasetTransformer:
             other_data=frame.other_data,
         )
 
-    def __change_point_coordinate_system(self, point: Union[Point, Point3D]):
+    def __change_point_coordinate_system(
+        self, point: Union[Point, Point3D, None]
+    ) -> Union[Point, Point3D, None]:
 
         if not point:
             return None
 
-        x = self._from_coordinate_system.pitch_dimensions.x_dim.to_base(
-            point.x
-        )
-        y = self._from_coordinate_system.pitch_dimensions.y_dim.to_base(
-            point.y
-        )
+        x = self._from_pitch_dimensions.x_dim.to_base(point.x)
+        y = self._from_pitch_dimensions.y_dim.to_base(point.y)
 
         if (
             self._from_coordinate_system.vertical_orientation
@@ -228,8 +248,8 @@ class DatasetTransformer:
             y = 1 - y
 
         if not self._to_coordinate_system.normalized:
-            x = self._to_coordinate_system.pitch_dimensions.x_dim.from_base(x)
-            y = self._to_coordinate_system.pitch_dimensions.y_dim.from_base(y)
+            x = self._to_pitch_dimensions.x_dim.from_base(x)
+            y = self._to_pitch_dimensions.y_dim.from_base(y)
 
         if isinstance(point, Point3D):
             return Point3D(x=x, y=y, z=point.z)
@@ -319,41 +339,36 @@ class DatasetTransformer:
 
         return replace(event, **position_changes)
 
-    def get_to_coordinate_system(self) -> CoordinateSystem:
+    def get_to_coordinate_system(self) -> Optional[CoordinateSystem]:
         return self._to_coordinate_system
 
     @classmethod
     def transform_dataset(
         cls,
         dataset: Dataset,
-        to_pitch_dimensions: PitchDimensions = None,
-        to_orientation: Orientation = None,
-        to_coordinate_system: CoordinateSystem = None,
+        to_pitch_dimensions: Optional[PitchDimensions] = None,
+        to_orientation: Optional[Orientation] = None,
+        to_coordinate_system: Optional[CoordinateSystem] = None,
     ) -> Dataset:
 
-        if to_pitch_dimensions and to_coordinate_system:
-            raise ValueError(
-                "You can't do both a PitchDimension and CoordinateSysetm on the same dataset transformation"
-            )
-
         if (
-            not to_pitch_dimensions
-            and not to_orientation
-            and not to_coordinate_system
+            to_pitch_dimensions is None
+            and to_orientation is None
+            and to_coordinate_system is None
         ):
             return dataset
-        elif not to_orientation:
-            to_orientation = dataset.metadata.orientation
 
-        if to_orientation == Orientation.BALL_OWNING_TEAM:
+        if to_orientation is None:
+            to_orientation = dataset.metadata.orientation
+        elif to_orientation == Orientation.BALL_OWNING_TEAM:
             if not dataset.metadata.flags & DatasetFlag.BALL_OWNING_TEAM:
                 raise ValueError(
-                    "Cannot transform to BALL_OWNING_TEAM orientation when dataset doesn't contain "
-                    "ball owning team data"
+                    "Cannot transform to BALL_OWNING_TEAM orientation when "
+                    "dataset doesn't contain ball owning team data"
                 )
 
-        if to_pitch_dimensions:
-
+        if to_pitch_dimensions is not None:
+            # Transform the pitch dimensions and optionally the orientation
             transformer = cls(
                 from_pitch_dimensions=dataset.metadata.pitch_dimensions,
                 from_orientation=dataset.metadata.orientation,
@@ -366,8 +381,8 @@ class DatasetTransformer:
                 orientation=to_orientation,
             )
 
-        elif to_coordinate_system:
-
+        elif to_coordinate_system is not None:
+            # Transform the coordinate system and optionally the orientation
             transformer = cls(
                 from_coordinate_system=dataset.metadata.coordinate_system,
                 from_orientation=dataset.metadata.orientation,
@@ -382,13 +397,26 @@ class DatasetTransformer:
             )
 
         else:
-
-            transformer = cls(
-                from_coordinate_system=dataset.metadata.coordinate_system,
-                from_orientation=dataset.metadata.orientation,
-                to_coordinate_system=dataset.metadata.coordinate_system,
-                to_orientation=to_orientation,
-            )
+            # Only transform the orientation
+            if dataset.metadata.coordinate_system is not None:
+                transformer = cls(
+                    from_coordinate_system=dataset.metadata.coordinate_system,
+                    from_orientation=dataset.metadata.orientation,
+                    to_coordinate_system=dataset.metadata.coordinate_system,
+                    to_orientation=to_orientation,
+                )
+            elif dataset.metadata.pitch_dimensions is not None:
+                transformer = cls(
+                    from_pitch_dimensions=dataset.metadata.pitch_dimensions,
+                    from_orientation=dataset.metadata.orientation,
+                    to_pitch_dimensions=dataset.metadata.pitch_dimensions,
+                    to_orientation=to_orientation,
+                )
+            else:
+                raise ValueError(
+                    "Cannot transform orientation when the dataset doesn't "
+                    "contain the pitch dimensions or a coordinate system"
+                )
             metadata = replace(
                 dataset.metadata,
                 orientation=to_orientation,

--- a/kloppy/helpers.py
+++ b/kloppy/helpers.py
@@ -1,4 +1,5 @@
-from typing import TypeVar, Union
+from typing import Union, Optional
+from collections.abc import Sequence
 
 from .domain import (
     Dataset,
@@ -12,47 +13,45 @@ from .domain import (
 )
 
 
-DatasetT = TypeVar("DatasetT")
-
-
 def transform(
     dataset: Dataset,
-    to_orientation=None,
-    to_pitch_dimensions=None,
-    to_coordinate_system: Union[CoordinateSystem, Provider] = None,
+    to_orientation: Optional[Union[Orientation, str]] = None,
+    to_pitch_dimensions: Optional[Union[PitchDimensions, Sequence]] = None,
+    to_coordinate_system: Optional[
+        Union[CoordinateSystem, Provider, str]
+    ] = None,
 ) -> Dataset:
-
-    if to_pitch_dimensions and to_coordinate_system:
-        raise ValueError(
-            "You can't do both a PitchDimension and CoordinateSysetm on the same dataset transformation"
-        )
-
-    if to_orientation and isinstance(to_orientation, str):
+    # convert raw orientation to object
+    if to_orientation is not None and isinstance(to_orientation, str):
         to_orientation = Orientation[to_orientation.upper()]
 
-    if to_pitch_dimensions and (
-        isinstance(to_pitch_dimensions, list)
-        or isinstance(to_pitch_dimensions, tuple)
+    # convert raw pitch dimensions to object
+    if to_pitch_dimensions is not None and isinstance(
+        to_pitch_dimensions, Sequence
     ):
         to_pitch_dimensions = PitchDimensions(
             x_dim=Dimension(*to_pitch_dimensions[0]),
             y_dim=Dimension(*to_pitch_dimensions[1]),
         )
-        return DatasetTransformer.transform_dataset(
-            dataset=dataset,
-            to_orientation=to_orientation,
-            to_pitch_dimensions=to_pitch_dimensions,
-        )
 
-    if to_coordinate_system and isinstance(to_coordinate_system, Provider):
-        to_coordinate_system = build_coordinate_system(
-            provider=to_coordinate_system,
-            length=dataset.metadata.coordinate_system.length,
-            width=dataset.metadata.coordinate_system.width,
-        )
+    # convert raw coordinate system to object
+    if to_coordinate_system is not None:
+        if isinstance(to_coordinate_system, str):
+            to_coordinate_system = build_coordinate_system(
+                provider=Provider[to_coordinate_system.upper()],
+                length=dataset.metadata.coordinate_system.length,
+                width=dataset.metadata.coordinate_system.width,
+            )
+        elif isinstance(to_coordinate_system, Provider):
+            to_coordinate_system = build_coordinate_system(
+                provider=to_coordinate_system,
+                length=dataset.metadata.coordinate_system.length,
+                width=dataset.metadata.coordinate_system.width,
+            )
 
     return DatasetTransformer.transform_dataset(
         dataset=dataset,
         to_orientation=to_orientation,
         to_coordinate_system=to_coordinate_system,
+        to_pitch_dimensions=to_pitch_dimensions,
     )

--- a/kloppy/tests/test_helpers.py
+++ b/kloppy/tests/test_helpers.py
@@ -129,6 +129,41 @@ class TestHelpers:
             )
         )
 
+    def test_transform_to_orientation(self):
+        tracking_data = self._get_tracking_dataset()
+
+        transformed_dataset = tracking_data.transform(
+            to_orientation=Orientation.AWAY_TEAM,
+        )
+        assert transformed_dataset.frames[0].ball_coordinates == Point3D(
+            x=0, y=50, z=0
+        )
+        assert (
+            transformed_dataset.metadata.orientation == Orientation.AWAY_TEAM
+        )
+
+    def test_transform_to_pitch_dimensions(self):
+        tracking_data = self._get_tracking_dataset()
+
+        transformed_dataset = tracking_data.transform(
+            to_pitch_dimensions=PitchDimensions(
+                x_dim=Dimension(min=0, max=1), y_dim=Dimension(min=0, max=1)
+            ),
+        )
+
+        assert transformed_dataset.frames[0].ball_coordinates == Point3D(
+            x=1, y=0, z=0
+        )
+        assert transformed_dataset.frames[1].ball_coordinates == Point3D(
+            x=0, y=1, z=1
+        )
+        assert (
+            transformed_dataset.metadata.pitch_dimensions
+            == PitchDimensions(
+                x_dim=Dimension(min=0, max=1), y_dim=Dimension(min=0, max=1)
+            )
+        )
+
     def test_transform_to_coordinate_system(self):
         base_dir = os.path.dirname(__file__)
 

